Zhuk Artsiom

B2etle opened the pull request.

Commit:
Add .gitkeep

Add empty .gitkeep file

Commit:
Implemented first and second iterations

Add one-shot command-line RSS reader application and set up distribution for it.

Commit:
Implemented third iteration

Added cache storage, which allows to cache news from feed and them from storage if necessary

Commit:
Implemented fourth iteration

Add posibility to convert news to HTML and PDF formats

Commit:
Implemented fifth iteration

Add posibility to print text result in colorized mode

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,26 @@
+from rss_reader import config
+
+
+setup(
```
После установки пакета в докер контейнере с пайтоном падает вот такая ошибка: 
```bash
# rss-reader --help
Traceback (most recent call last):
  File "/usr/local/bin/rss-reader", line 5, in <module>
    from rss_reader.rss_reader import run
  File "/usr/local/lib/python3.8/site-packages/rss_reader/rss_reader.py", line 10, in <module>
    from rss_reader.cache_storage import ReadCache, WriteCache
  File "/usr/local/lib/python3.8/site-packages/rss_reader/cache_storage.py", line 8, in <module>
    import tinydb
ModuleNotFoundError: No module named 'tinydb'
```

B2etle replied:
Добавил requirements в setup.py

AlexeiBuzuma commented on rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,165 @@
+"""Application main module."""
```
Часто выпадают ошибки 
```bash 
# rss-reader "https://news.yahoo.com/rss/" --json --verbose
[11.25.2019 21:52:19 | rss_reader]: Successful initialze RSS reader
[11.25.2019 21:52:19 | rss_reader]: Application arguments: source = https://news.yahoo.com/rss/; json = True; verbose = True; limit = None
[11.25.2019 21:52:19 | rss_reader]: Run RSS reader
[11.25.2019 21:52:19 | rss_reader]: Data obtain has been started
[11.25.2019 21:52:20 | source_parser]: Successful get RSS data from RSS URL
[11.25.2019 21:52:20 | source_parser]: Start parse RSS source data
[11.25.2019 21:52:20 | source_parser]: RSS news has been parsed successfuly
[11.25.2019 21:52:20 | rss_reader]: Data obtain has been finished
[11.25.2019 21:52:20 | rss_reader]: Start display news from a RSS URL
[11.25.2019 21:52:20 | rss_reader]: __init__() takes 2 positional arguments but 3 were given
RSS reader has beed completed unsuccessfully
```

B2etle replied:
Убрал ненужный параметр при инициализации печати в JSON формате.

Commit:
Add requiremens to setup.py

Add property install_requires with all required packages to setup.py.

Commit:
Fix JSON output problem

Remove passing an unnecessary parameter to the DisplayNewsJson class

Commit:
Сorrect work with paths

Working with paths is implemented through the libraries intended for this purpose

Commit:
Correct JSON validation

Correct json validation when displaying news in json format

AlexeiBuzuma commented on rss_reader/cache_storage.py:
```diff
@@ -0,0 +1,109 @@
+            try:
+                self.verify_data(news_data)
+            except RSSNewsCacheError:
+                pass
```
Если исключение нужно просто пропустить, то есть смысл хотя бы залогировать какое-нибудь дебажное сообщенеие, что тут вот что-то было не так. В будущем это может помощь при отладке.

B2etle replied:
Переделал логику функции verify_data, убрал исключение.

Commit:
Add unit tests

Add unit tests to rss-reader application

Commit:
Update unit tests

Update unit tests for rss-reader application
